### DISCUSSION TITLE
Multiple Frigates.. One WEB ui?

### DISCUSSION THREAD

**Discussion by Minglarn** [blakeblackshear/frigate#1690] [Ideas]
I could have a brain meltdown after a hard day at work, but I just attached 2 more cameras to my i5 server, and yes, it struggles...
Then a fly flew by and sat on my lenovo laptop with a broken screen and I got an idea. 
Could my lenovo act as a second Frigate instance?
Installed ubuntu 20 LT on that machine as well and installed Frigate. 
It's up and running with 2 new Reolink cams but it seeams that my second Frigate cant have the same MQTT topic. 
This means that I've to watch 2 different MQTT's in HA for notifications, and 2 different frigate webpages to watch the events. 

It would be super if you could combine 2 or more instances of Frigate to act as 1.
Consider this:
1 webpage for all instances where ALL cams would be visible under. 
1 proxy (if its possible, really don't know)...
1 MQTT topic where all cams could broadcast to. 

Or is it just me that makes it complicated? Tried to search for this type of setup but did not find any about this in the discussions.

**Comment by blakeblackshear** [comment]
The home assistant integration already supports multiple instances, but they are merged completely. Creating a single UI for a distributed system would be a major architectural overhaul, and one that would benefit very few.

**Comment by Laggy47-alt** [comment]
I have been experimenting with frigate and MQTT and have hit a brick wall.
I have 5 instances of frigate and 1 MQTT broker.
I can send data over MQTT from all 5 instances and receive them as site1,site2,site3,site4,site5
my problem is- i want to integrate all 5 sites with a single dashboard for alert viewing purely and i am not succeeding in any way
is there no way for frigate to subscribe to a broker that send the frigate/event topic and add them in the alert section?

**Reply by Laggy47-alt** [reply]
sorry i forgot to mention that home assistant does not have a dashboard like frigate does.

**Reply by hawkeye217** [reply]
No, Frigate's UI is designed around a single instance of Frigate only. It would be a massive refactor of almost every part of the code to support multiple instances.

**Reply by ZhaiSoul** [reply]
Perhaps we could consider using a proxy to merge all Frigate instances into a single endpoint, which would be read-only and not allow configuration changes. This approach should require minimal development effort and would suit most people who need multiple instances.

**Reply by Laggy47-alt** [reply]
<img width="1659" height="271" alt="Screenshot From 2026-05-13 10-42-52" src="https://github.com/user-attachments/assets/e6d45358-7f93-4306-a001-5790569ae3c5" />
i created a webapp that can accept multiple frigate instances and display alerts in one unified wall for acknowledgement tested it with 5 NVR instances and still running great......though it takes a lot of my capacity to work on it so it will probably require funding in some way - however i do have a coupon option that can enable lifetime accessibility. 
it does require some setup though but overall it works very well

**Comment by Laggy47-alt** [comment]
I actually got it working.
Haha I'll be testing it out and once done provide some feedback.

Kind regards.

**Comment by Laggy47-alt** [comment]
I would lobe to pick tour brain about suitable server for the following
site configs.

Small - 4-8 cameras
Meduim - 8-16
Enterprise 16-64+ ( i would think you would need multiple devices or 1 bug
server with dual xeon procs or simmilar?)

Kind regards.

**Comment by meeki007** [comment]
This does not solve your mqtt but my solution to running multiple frigate servers is node-red worldmap.
see this frigate discussions --->  https://github.com/blakeblackshear/frigate/issues/14858#issuecomment-2675461156
When you click a camera on the map it sends you to the firgate server the camera is on.
I also use node-red for my mqtt combinations
